lib/opensusebasetest.pm

```diff
@@ -556,13 +556,58 @@ sub remount_tmp_if_ro {
     script_run 'touch /tmp/test_ro || mount -t tmpfs /dev/shm /tmp';
 }
 
+=head2 select_serial_terminal
+
+    select_serial_terminal($root);
+
+Select most suitable text console with root user.
+The choice is made by BACKEND and other variables.
+
+Optional C<root> parameter specifies, whether use root user
+(1 or default) or prefer non-root user if available.
+=cut
+sub select_serial_terminal {
+    my ($self, $root) = @_;
+    $root //= 1;
+
+    my $backend = get_required_var('BACKEND');
+    my $console;
+
+    if ($backend eq 'qemu') {
+        if (get_var('NO_VIRTIO_CONSOLE')) {
+            $console = $root ? 'root-console' : 'user-console';
+        } else {
+            $console = $root ? 'root-virtio-terminal' : 'virtio-terminal';
+        }
+    } elsif (get_var('S390_ZKVM')) {
+        $console = $root ? 'root-console' : 'user-console';
+    } elsif ($backend =~ /^(ikvm|ipmi|spvm)$/) {
+        $console = 'root-ssh';
+    }
+
+    die "No support for backend '$backend', add it" if ($console eq '');
+    bmwqemu::fctinfo("Selecting console '$console'");
```

Comment:
not sure what actually `fctinfo` is doing but isn't this line overlap with this https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L1430 ?

Reply:
Whole output in log http://quasar.suse.cz/tests/1188/file/autoinst-log.txt :+1: 
```
[2018-10-30T15:33:45.0876 CET] [info] ::: opensusebasetest::select_serial_terminal: Selecting console 'root-virtio-terminal'
[2018-10-30T15:33:45.0876 CET] [debug] /var/lib/openqa/share/tests/sle/tests/hpc/before_test.pm:22 called opensusebasetest::select_serial_terminal
[2018-10-30T15:33:45.0876 CET] [debug] <<< testapi::select_console(testapi_console='root-virtio-terminal')
```
I wanted it to make clear where error happen, but maybe it's really too verbose, so I'll remove it.